idioms/ffi-accepting-strings.md

```diff
@@ -22,15 +34,23 @@ pub mod unsafe_module {
 
     // other module content
 
+    /// Log a message at the specified level.
+    ///
+    /// # Safety
+    ///
+    /// It is the caller's guarantee to ensure `msg`
+    ///
+    /// - is not a null pointer
+    /// - points to valid, initialized data
+    /// - points to memory ending in a null byte
+    /// - won't be mutated for the duration of this function call
     #[no_mangle]
-    pub extern "C" fn mylib_log(msg: *const libc::c_char, level: libc::c_int) {
+    pub unsafe extern "C" fn mylib_log(msg: *const libc::c_char, level: libc::c_int) {
```

Comment:
I don't like that we are extending the scope of unsafe.
`crate::log` is a safe function, so there is no point in placing it inside unsafe, right?
Maybe we could extract `match` into its own unsafe function?

Reply:
The reason is because [`#[no_mangle]` is inherently unsafe, due to implied guarantees about the rest of the code](https://github.com/rust-lang/rust/issues/28179). It is those guarantees that the `unsafe fn` represents.

Reply:
Thank you, I learned something :)
Marking this as solved.